web/src/unread_ops.ts

```diff
@@ -336,6 +374,9 @@ function do_mark_unread_by_narrow(
             } else if (loading_indicator_displayed) {
                 finish_loading(messages_marked_unread_till_now);
             }
+            if (ignored_because_not_subscribed.length > 0) {
+                handle_skipped_unsubscribed_streams(ignored_because_not_subscribed);
```

Comment:
How does this work when this code path is called recursively (see the FOLLOWUP_BATCH_SIZE logic)? 

Probably one wants to accumulate a set of channels by merging what was returned with the set we had previously, passing into the recursive calls, and only do this call from inside the block that has `finish_loading`.